framework/helpers/BaseUrl.php

```diff
@@ -378,7 +371,20 @@ public static function home($scheme = false)
      */
     public static function isRelative($url)
     {
-        return strncmp($url, '//', 2) && strpos($url, '://') === false;
+        $parts = parse_url($url, PHP_URL_HOST);        
+        return empty($parts);
+    }
+
+    /**
+     * Returns a value indicating whether a URL is protocol agnostic.
+     * @param string $url the URL to be checked
+     * @return bool whether the URL is protocol agnostic
+     * @since 2.0.16.2
+     */
+    public static function isProtocolAgnostic($url)
+    {
+        $parts = parse_url($url, PHP_URL_SCHEME);
```

Comment:
This method will return `true` for `example.com/test`, which IMO is incorrect - I would expect `true` only from URLs starting with `//`.

Reply:
The idea of the method is to return true for relative URLs as far as I understand.

Reply:
@SamMousa ?

Reply:
> The idea of the method is to return true for relative URLs as far as I understand.

This looks unpractical for me, because relative URLs and protocol-relative URLs are completely different things and should be processed differently (you can't just add protocol to relative URL). This method is useless on its own - you always need to use it in pair with `isRelative()`. And it make sense only if `isRelative()` returns `false`, since there is no protocol for relative URL, you don't need `isProtocolAgnostic()` to known that. :)

Reply:
Hmm... thinking of it, I can't find a use case for the method by itself. @SamMousa what was the initial idea?

Reply:
You always combine the calls, but that doesn't make a difference.
If it returns `false` for a relative URL, you will still need to check why it is returning `false` (since this could mean that it is a relative URL or an absolute URL without a scheme..

So from that perspective instead of creating a compound check, the function does something that is simple to understand: it answers the question of whether this URL includes a scheme or not.

Reply:
I may want to check if URL is protocol-relative (`//example.com`):

```php
return Url::isProtocolRelative($url) ? "https:$url" : $url;
```

If we leave old implementation (`strncmp($url, '//', 2) === 0`) method will we also useful as standalone method, not only as appendix for `isRelative()`.

Reply:
We have three possible cases and 2 potential implementations of `isProtocolRelative()`:

| Case | `empty(parse_url($url, PHP_URL_SCHEME))` | `substr($url, 0, 2) === '//'` |
| ---- | ----- | ---- |
| Is relative URL (`/path`) | `Url::isRelative($url)` | `Url::isRelative($url)` |
| Is protocol-relative URL (`//exmaple.com/path`) | `Url::isRelative($url) && Url::isProtocolRelative($url)` | `Url::isProtocolRelative($url)` |
| Is absolute URL (`https://exmaple.com/path`) | `!Url::isProtocolRelative($url)` | `!Url::isRelative($url) && !Url::isProtocolRelative($url)` |

So maybe instead of `isProtocolRelative()` we should implement `isAbsolute()` as `!empty(parse_url($url, PHP_URL_SCHEME))`? Testing whether URL is absolute is the only case when `parse_url($url, PHP_URL_SCHEME)` check may be used on its own. Also it seems to more common and easier to understand than protocol relative URLs.